File: src/Panorama.Core/Scenes/Events/ScenesRetrieved/ScenesRetrievedEventData.cs
using Newtonsoft.Json;
using Panorama.Backing.Bus.Shared.Common.Dto;
using Panorama.Backing.Bus.Shared.Scenes.Dto;
using Panorama.Events.Base;

namespace Panorama.Scenes.Events.ScenesRetrieved;

public class ScenesRetrievedEventData : AppEventData
{
    [JsonProperty("data")]
    public PagedResultDto<ViewSceneDto> Data { get; set; }
}

File: src/Panorama.Web.Host/src/shared/service-proxies/scenography/events/ScenesRetrievedEventData.ts
import {ViewSceneDto} from "../dtos/ViewSceneDto";
import {PagedResultDto} from "../../common/dtos/PagedResultDto";

export class ScenesRetrievedEventData {
    data: PagedResultDto<ViewSceneDto>;
    
    constructor(data: Partial<ScenesRetrievedEventData> = {}) {
        this.data = data.data || undefined;
    }
}